src/Service/Resolvers/Sql Query Structures/SqlInsertQueryStructure.cs

```diff
@@ -74,6 +74,15 @@
                 MetadataProvider.TryGetBackingColumn(EntityName, param.Key, out string? backingColumn);
                 PopulateColumnsAndParams(backingColumn!, param.Value);
             }
+
+            if (FieldsReferencedInDbPolicyForCreateAction.Count > 0)
+            {
+                // This indicates that one or more fields referenced in the database policy are not a part of the insert statement.
+                throw new DataApiBuilderException(
+                    message: "One or more fields referenced by the database policy are not present in the request body.",
```

Comment:
Does this work for graphql? 

Have we tested create policy for graphql?

Reply:
Yes it works for graphql as well. In process of writing the test case for that.